modules/core/src/main/java/org/apache/ignite/internal/properties/IgniteProductVersion.java

```diff
@@ -29,11 +30,9 @@
  */
 public class IgniteProductVersion implements Serializable {
     private static final Pattern VERSION_PATTERN =
-            Pattern.compile("(?<major>\\d+)\\.(?<minor>\\d+)\\.(?<maintenance>\\d+)((?<snapshot>-SNAPSHOT)|-(?<alpha>alpha\\d+))?");
+            Pattern.compile("(?<major>\\d+)\\.(?<minor>\\d+)\\.(?<maintenance>\\d+)(\\.(?<patch>\\d+))?(-(?<preRelease>[0-9A-Za-z]+))?");
```

Comment:
Can you please add a description about the format to the javadoc? E.g. what `preRelease` is and some examples for Ignite versions?

Reply:
Can we replace `[0-9A-Za-z]` with `\w`?

Reply:
From [docs](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/regex/Pattern.html) \w - A word character: [a-zA-Z_0-9] which doesn't quite fit

Reply:
why? Are you worried about the underscore?

Reply:
According to the [SemVer2](https://semver.org/#spec-item-9) pre release cannot contain underscores.

Reply:
Looks awesome!

Reply:
Fix it